kratos/processes/calculate_discontinuous_distance_to_skin_process.cpp

```diff
@@ -36,19 +37,35 @@ namespace Kratos
 		: mFindIntersectedObjectsProcess(rVolumePart, rSkinPart)
 		, mrSkinPart(rSkinPart)
 		, mrVolumePart(rVolumePart)
-		, mOptions(CalculateDiscontinuousDistanceToSkinProcessFlags::CALCULATE_ELEMENTAL_EDGE_DISTANCES.AsFalse())
+		, mOptionEdge(CalculateDiscontinuousDistanceToSkinProcessFlags::CALCULATE_ELEMENTAL_EDGE_DISTANCES.AsFalse())
+		, mOptionExtraEdge(CalculateDiscontinuousDistanceToSkinProcessFlags::CALCULATE_ELEMENTAL_EXTRAPOLATED_EDGE_DISTANCES.AsFalse())
 	{
 	}
 
 	template<std::size_t TDim>
 	CalculateDiscontinuousDistanceToSkinProcess<TDim>::CalculateDiscontinuousDistanceToSkinProcess(
 		ModelPart& rVolumePart,
 		ModelPart& rSkinPart,
-		const Flags rOptions)
+		const Flags rOptionEdge)
 		: mFindIntersectedObjectsProcess(rVolumePart, rSkinPart)
 		, mrSkinPart(rSkinPart)
 		, mrVolumePart(rVolumePart)
-		, mOptions(rOptions)
+		, mOptionEdge(rOptionEdge)
+		, mOptionExtraEdge(CalculateDiscontinuousDistanceToSkinProcessFlags::CALCULATE_ELEMENTAL_EXTRAPOLATED_EDGE_DISTANCES.AsFalse())
+	{
+	}
+
+	template<std::size_t TDim>
+	CalculateDiscontinuousDistanceToSkinProcess<TDim>::CalculateDiscontinuousDistanceToSkinProcess(
+		ModelPart& rVolumePart,
+		ModelPart& rSkinPart,
+		const Flags rOptionEdge,
+		const Flags rOptionExtraEdge)
+		: mFindIntersectedObjectsProcess(rVolumePart, rSkinPart)
+		, mrSkinPart(rSkinPart)
+		, mrVolumePart(rVolumePart)
+		, mOptionEdge(rOptionEdge)
+		, mOptionExtraEdge(rOptionExtraEdge)
```

Comment:
I wouldn't create an extra constructor. Note that the same behavior can be achieved with the flag ones above.

Reply:
Indeed, we lack a `Model` and `KratosParameters` constructor. If you want to add a new one, lets take the chance to add this.

Reply:
If there isn't an additional constructor the user always needs to provide no flag or both flags and then the second flag kind of fails it purpose, doesn't it?

Reply:
Maybe I am miss-using the entire flag thing ..

Reply:
(see comment in the header)